x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingDeciderServiceTests.java

```diff
@@ -213,11 +213,17 @@ public void testScaleDown_WhenMemoryIsInaccurate() {
 
         MlAutoscalingDeciderService service = buildService();
         MlScalingReason.Builder reasonBuilder = new MlScalingReason.Builder();
-        assertThat(service.checkForScaleDown(nodes,
-            ClusterState.EMPTY_STATE,
-            Long.MAX_VALUE,
-            new NativeMemoryCapacity(ByteSizeValue.ofGb(3).getBytes(), ByteSizeValue.ofGb(1).getBytes()),
-            reasonBuilder).isEmpty(), is(true));
+        try {
+            service.checkForScaleDown(nodes,
+                ClusterState.EMPTY_STATE,
+                Long.MAX_VALUE,
+                new NativeMemoryCapacity(ByteSizeValue.ofGb(3).getBytes(), ByteSizeValue.ofGb(1).getBytes()),
+                reasonBuilder);
+        } catch (AssertionError ae) {
+            // the scale method should thrown an assertion failure
+            return;
+        }
+        assert false : "call for scale down should have thrown an assertion error";
```

Comment:
Does `expectThrows` work with `AssertionError`?  If it does then that would be the more idiomatic way to write this.

Reply:
I can try that, but iirc, it doesn't as expectThrows has an assertion in it itself.